cmd/pcap_converter/main.go

```diff
@@ -6,36 +6,140 @@ import (
 	"fmt"
 	"log"
 	"os"
+	"reflect"
 	"runtime"
 	"strings"
+	"time"
 
+	"github.com/ParsePlatform/flashback"
 	"github.com/google/gopacket/pcap"
-	"github.com/mongodb/mongo-tools/common/bsonutil"
-	"github.com/mongodb/mongo-tools/common/json"
 	"github.com/tmc/mongocaputils"
 	"github.com/tmc/mongoproto"
 	"gopkg.in/mgo.v2/bson"
 )
 
 var (
-	pcapFile        = flag.String("f", "-", "pcap file (or '-' for stdin)")
+	pcapFile	= flag.String("f", "-", "pcap file (or '-' for stdin)")
+	bsonFile	= flag.String("o", "flashback.bson", "bson output file, will be overwritten")
 	packetBufSize   = flag.Int("size", 1000, "size of packet buffer used for ordering within streams")
 	continueOnError = flag.Bool("continue_on_error", false, "Continue parsing lines if an error is encountered")
+	debug           = flag.Bool("debug", false, "Print debug-level output")
 )
 
-// Converts the raw BSON doc in op.Query to extended JSON
-func rawBSONToJSON(rawBSON []byte) (interface{}, error) {
-	// Use bson.D to preserve order when unmarshalling
-	// http://godoc.org/github.com/mongodb/mongo-tools/common/bsonutil#MarshalD
-	var data bson.D
-	if err := bson.Unmarshal(rawBSON, &data); err != nil {
-		return nil, err
+type Operation struct {
```

Comment:
That's what I tried originally but it ended up writing extra fields to the bson file because optional/conditional fields in the flashback.Op struct do not have "omitempty". This caused every possible struct field to be written to the bson file for every op even if unset/unrelated. An empty QueryDoc or nToSkip would show up on insert operations, for example.

I think 'flashback' would probably parse that format fine but I decided to match the output to how record.py does it mostly to save space.